test/unit/org/apache/cassandra/index/sai/metrics/QueryMetricsTest.java

```diff
@@ -594,6 +594,91 @@ private void testQueryKindMetrics(boolean perTable, boolean perQuery)
         waitForHistogramCountEqualsIfExists(perQuery, objectName(name, PER_MP_HYBRID_QUERY_METRIC_TYPE), 1);
     }
 
+    @Test
+    public void testQueryPlannerMetrics()
+    {
+        String table = createTable("CREATE TABLE %s (k int PRIMARY KEY, lc int, hc int)");
+        createIndex("CREATE CUSTOM INDEX ON %s(lc) USING 'StorageAttachedIndex'");
+        createIndex("CREATE CUSTOM INDEX ON %s(hc) USING 'StorageAttachedIndex'");
+
+        int numRows = 10000;
+        for (int i = 0; i < numRows; i++)
+        {
+            execute("INSERT INTO %s (k, lc, hc) VALUES (?, ?, ?)", i, i % 2, i);
+        }
+
+        flush();
+
+        UntypedResultSet rows = execute("SELECT k FROM %s WHERE lc = 0");
+        assertEquals(numRows / 2, rows.size());
+
+        var rowsToReturnEstimatedMetric = objectNameNoIndex("RowsToReturnEstimated", KEYSPACE, table, PER_QUERY_METRIC_TYPE);
+        waitForHistogramCountEquals(rowsToReturnEstimatedMetric, 1);
+        waitForHistogramMeanBetween(rowsToReturnEstimatedMetric, numRows / 2.0 * 0.75, numRows / 2.0 * 1.25);
+
+        var rowsToFetchEstimatedMetric = objectNameNoIndex("RowsToFetchEstimated", KEYSPACE, table, PER_QUERY_METRIC_TYPE);
+        waitForHistogramCountEquals(rowsToFetchEstimatedMetric, 1);
+        waitForHistogramMeanBetween(rowsToFetchEstimatedMetric, numRows / 2.0 * 0.75, numRows / 2.0 * 1.25);
+
+        var keysToIterateEstimatedMetric = objectNameNoIndex("KeysToIterateEstimated", KEYSPACE, table, PER_QUERY_METRIC_TYPE);
+        waitForHistogramCountEquals(keysToIterateEstimatedMetric, 1);
+        waitForHistogramMeanBetween(keysToIterateEstimatedMetric, numRows / 2.0 * 0.75, numRows / 2.0 * 1.25);
+
+        var objectName = objectNameNoIndex("CostEstimated", KEYSPACE, table, PER_QUERY_METRIC_TYPE);
+        waitForHistogramCountEquals(objectName, 1);
+        waitForHistogramMeanBetween(objectName, 1.0, Double.POSITIVE_INFINITY);
+
+        var logSelectivityEstimatedMetric = objectNameNoIndex("LogSelectivityEstimated", KEYSPACE, table, PER_QUERY_METRIC_TYPE);
+        waitForHistogramCountEquals(logSelectivityEstimatedMetric, 1);
+        waitForHistogramMeanBetween(logSelectivityEstimatedMetric, 0, 0);
+
+        objectName = objectNameNoIndex("IndexReferencesInQuery", KEYSPACE, table, PER_QUERY_METRIC_TYPE);
+        waitForHistogramCountEquals(objectName, 1);
+        waitForHistogramMeanBetween(objectName, 1.0, 1.0);
+
+        objectName = objectNameNoIndex("IndexReferencesInPlan", KEYSPACE, table, PER_QUERY_METRIC_TYPE);
```

Comment:
Nit: we can reuse this `ObjectName`, same as it's done for `logSelectivityEstimatedMetric`:
```
var indexReferencesInPlanMetric = objectNameNoIndex("IndexReferencesInPlan", KEYSPACE, table, PER_QUERY_METRIC_TYPE);
```